gnovm/pkg/gnolang/preprocess.go

```diff
@@ -3073,6 +3073,237 @@ func findUndefined(store Store, last BlockNode, x Expr) (un Name) {
 	return findUndefined2(store, last, x, nil)
 }
 
+// finds the next undefined identifier and returns it if it is global
+func findUndefined2SkipLocals(store Store, last BlockNode, x Expr, t Type) Name {
```

Comment:
please consider this:
```go
package main

var myDep string

var myVar2 = func() {
	aaa := ""
	myDep2 := "yeah"
	switch aaa {
	case myDep:
		println(myDep2)
	}
}

var myDep2 = "hey"

func main() {
	myVar2()
}
```
in the current situation, while dealing with `switch aaa`,  `aaa` is deemed `un` -> not global -> not local(at the moment the previous `aaa := ""` has not been preprocessed(but it's been `predefined`, so the name exists in block).

the situation is same for `myDep2`, when dealing with `println(myDep2)`, the global value declaration `var myDep2 = "hey"` is conducted even it's shadowed by the local `myDep2 := "yeah"`.  the reason is all the same, at this moment, the XXX is not global nor local.

I think the potential way to improve is to modify `findUndefined2`, to return another result to represent if it's local or not.

Reply:
This correctly prints `yeah`.

Reply:
The problem is the usage of `findUndefined2` here.

```go
func findUndefined2SkipLocals(store Store, last BlockNode, x Expr, t Type) Name {
	name := findUndefined2(store, last, x, t)
```